src/pkg/packager/deploy.go

```diff
@@ -216,6 +215,18 @@ func (p *Packager) deployInitComponent(component types.ZarfComponent) (charts []
 	// Before deploying the seed registry, start the injector
 	if isSeedRegistry {
 		p.cluster.StartInjectionMadness(p.layout.Base, p.layout.Images.Base, component.Images)
+
+		// Retrieve the default storage class from the cluster, it will be the default
+		if p.cfg.PkgOpts.SetVariables == nil {
+			p.cfg.PkgOpts.SetVariables = map[string]string{}
+		}
+		if c := p.cfg.PkgOpts.SetVariables["REGISTRY_STORAGE_CLASS"]; c == "" {
+			p.cfg.PkgOpts.SetVariables["REGISTRY_STORAGE_CLASS"], err = p.cluster.GetDefaultStorageClass()
+		}
+
+		if err != nil {
+			message.WarnErr(err, "Unable to get the default storage class from the cluster")
```

Comment:
I think we have an opportunity to fail fast here. If `REGISTRY_PVC_ENABLED` is true (this is the default), the user did not specify a storage class to use, and there was no default storage class found in the cluster, we should fail fast and notify the user that a storage class is needed for the registry to deploy successfully.

RKE2 clusters come to mind since they don't provide a default storage class